### PR TITLE
Make choice question data handling less string-y

### PR DESCRIPTION
**Pull request by inducer**
@dzhuang Could you give this a look, since it's an important code path?

### CODE REVIEW

**Comment by dzhuang** [comment]
I'll test that.

File: course/page/choice.py [hunk 1]
```diff
@@ -65,20 +64,78 @@ def markup_to_html_plain(page_context, s):
...
 
-# {{{ choice question base
+# {{{ choice data model
 
-class ChoiceQuestionBase(PageBaseWithTitle, PageBaseWithValue):
+class ChoiceModes:
+    INCORRECT = "incorrect"
+    CORRECT = "correct"
+    DISREGARD = "disregard"
+    ALWAYS_CORRECT = "always_correct"
```
Comment (dzhuang) [inline course/page/choice.py]: Can you explain the use case of "aways_correct"? Thanks.

Comment (inducer) [reply course/page/choice.py]: Sure. It exists to ensure fair scoring of a multi-select question in which one option has turned out to be flawed. The net effect of `disregard` is to score the question as if that option didn't exist. But some students may have received points from the broken option, so `disregard` would take those points away. Cue lots of (somewhat justified) complaints from grumpy students. `always_correct` prevents that by grading any answer as a correct one, therefore never leading to a point decrease.

If this explanation was helpful, could you copy-paste it to the docs? Thanks! :)

Comment (dzhuang) [reply course/page/choice.py]: Ok. Thanks.

Comment (dzhuang) [reply course/page/choice.py]: I've submit a PR in https://github.com/inducer/relate-sample/pull/10/files, can you review that?

File: course/page/choice.py [hunk 2]
```diff
@@ -65,20 +64,78 @@ def markup_to_html_plain(page_context, s):
...
+                            _("%(location)s: more than one choice mode set")
+                            % {'location': location})
+
+                item_mode[0] = mode
+
+            return s
+
+        s = mode_from_prefix(ChoiceModes.CORRECT, cls.CORRECT_TAG, s)
+        s = mode_from_prefix(ChoiceModes.DISREGARD, cls.DISREGARD_TAG, s)
+        s = mode_from_prefix(ChoiceModes.ALWAYS_CORRECT, cls.ALWAYS_CORRECT_TAG, s)
```
Comment (dzhuang) [inline course/page/choice.py]: This won't handle `~CORRECT~~CORRECT~foobar`, i.e., it will view `~CORRECT~foobar` as a correct answer.

Comment (inducer) [reply course/page/choice.py]: Should it?

Comment (dzhuang) [reply course/page/choice.py]: (sorry for late reponse) `~CORRECT~~DISREGARD~foobar` will raise `ValidationError`, am I right?

Comment (dzhuang) [reply course/page/choice.py]: And `~DISREGARD~~CORRECT~foobar` is also expected to raise a `ValidationError` but not.

Comment (inducer) [reply course/page/choice.py]: Agreed,

**Comment by inducer** [comment]
There are some merge conflicts in the tests. @dzhuang, could you take a look?

**Comment by dzhuang** [comment]
@inducer To resolve this, you need to merge #442 into this PR. and merge #441 into master. Then it can be resolved.

**Comment by inducer** [comment]
@dzhuang, you mentioned in #412 that it LGTY, so I'll go ahead and merge it.

**Comment by dzhuang** [comment]
Thank you！

**Pull request merged**